sdk/appconfiguration/azure-appconfigurationprovider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,217 @@
+from azure.appconfiguration import AzureAppConfigurationClient
+from azure.keyvault.secrets import (
+     SecretClient,
+     KeyVaultSecretIdentifier
+)
+from azure.core.exceptions import (
+    HttpResponseError,
+    ResourceNotFoundError
+)
+from ._settingselector import SettingSelector
+from ._azure_appconfiguration_provider_error import KeyVaultReferenceError
+from ._constants import KEY_VAULT_REFERENCE_CONTENT_TYPE
+
+from ._user_agent import USER_AGENT
+
+
+class AzureAppConfigurationProvider:
+    """
+    Provides a dictionary-like interface to Azure App Configuration settings. Enables loading of sets of configuration
+    settings from Azure App Configuration into a Python application. Enables trimming of prefixes from configuration
+    keys. Enables resolution of Key Vault references in configuration settings.
+    """
+
+    def __init__(self):
+        # type: () -> None
+        self._dict = {}
+        self._trim_prefixes = []
+        self._client = None
+
+    @classmethod
+    def load(cls, connection_string=None, endpoint=None, credential=None, **kwargs):
+        """
+        Loads configuration settings from Azure App Configuration into a Python application.
+
+        :param connection_string: Connection string (one of connection_string or endpoint and credential must be set)
+        :type connection_string: str
+        :param endpoint: Endpoint (one of connection_string or endpoint and credential must be set)
+        :type endpoint: str
+        :param credential: Credential (one of connection_string or endpoint and credential must be set)
+        :type credential: Union[AppConfigConnectionStringCredential, TokenCredential]
+        :keyword selectors: List of setting selectors to filter configuration settings
+        :type selectors: list[~azure.appconfigurationprovider.SettingSelector]
+        :keyword trim_prefixes: List of prefixes to trim from configuration keys
+        :type trim_prefixes: list[str]
+        :keyword key_vault_options: Options for resolving Key Vault references
+        :type key_vault_options: ~azure.appconfigurationprovider.KeyVaultOptions
+        """
+        provider = AzureAppConfigurationProvider()
+
+        key_vault_options = kwargs.pop("key_vault_options", None)
+
+        provider.__buildprovider(connection_string, endpoint,
+                                 credential, key_vault_options)
+
+        selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
+
+        provider._trim_prefixes = sorted(kwargs.pop(
+            "trimmed_key_prefixes", []), key=len, reverse=True)
+
+        provider._dict = {}
+
+        secret_clients = key_vault_options.secret_clients if key_vault_options else {}
+
+        for select in selects:
+            try:
+                configurations = provider._client.list_configuration_settings(
+                    key_filter=select.key_filter, label_filter=select.label_filter)
+            except HttpResponseError as e:
+                raise e
```

Comment:
Why are we explicitly re-raising here? Are we trying to control the traceback?

Reply:
I think this is a side effect of a planed `fail-fast` option, where it doesn't kill the load operation and instead would continue on and load everything else.

Reply:
We can remove this exception handling till we add this fail-fast option in future. For fail-fast, we'd still need be catching very specific HttpResponseError types, not all. So lets remove it for now.